pkg/resources/flow.go

```diff
@@ -87,11 +86,32 @@ func (f *Flow) Create() error {
 		}
 	}
 	options := interfaces.DependencyGraphOptions{
-		FlowName: f.originalName,
-		Args:     args,
+		FlowName:         f.originalName,
+		Args:             args,
+		FlowInstanceName: f.generatedName,
 	}
+
+	if destroy {
+		// Delete one replica
+		options.ReplicaCount = -1
```

Comment:
The only case when `Delete` is called is when user deletes a replica of some flow which triggers this flow during its construction. Thus `Delete` should be have a reverse effect from `Create` which was called during construction. `Create` creates one replica so `Delete` also deletes one. Imagine what would happen if I call flow X from both flows A and B and then delete replica of A. If X were to delete all its replicas B would become invalid